frontend/js/src/common/brainzplayer/NavidromePlayer.tsx

```diff
@@ -0,0 +1,429 @@
+      if (!authParams) {
+        handleWarning(
+          "Navidrome authentication not available. Please check your connection.",
+          "Authentication Error"
+        );
+        onTrackNotFound();
+        return;
+      }
+
+      const track = await searchForNavidromeTrack(
+        this.getNavidromeInstanceURL(),
+        authParams,
+        trackName,
+        artistName
+      );
+
+      if (track) {
+        this.setState({ currentTrack: track });
+        const audioElement = this.audioRef.current;
+        if (audioElement) {
+          const streamUrl = this.getNavidromeStreamUrl(track.id);
+          this.setAudioSrc(audioElement, streamUrl);
+          await audioElement.play();
+          return;
+        }
+      }
+
+      handleWarning(
+        `"${trackName}" by ${artistName} is not available on your Navidrome server`,
+        "Track not available on Navidrome"
+      );
+      onTrackNotFound();
+    } catch (errorObject) {
+      if (errorObject.status === 401) {
+        this.handleAuthenticationError();
+        return;
+      }
+      handleError(
+        errorObject.message ?? errorObject,
+        "Error searching on Navidrome"
+      );
+    }
+  };
+
+  handleAuthenticationError = (): void => {
+    const { onInvalidateDataSource } = this.props;
+
+    onInvalidateDataSource(
+      this as any,
+      <span>
+        Please{" "}
+        <Link to="/settings/music-services/details/">
+          re-connect your Navidrome account
+        </Link>
+      </span>
+    );
+  };
+
+  togglePlay = async (): Promise<void> => {
+    const audioElement = this.audioRef.current;
+    if (!audioElement) return;
+
+    try {
+      if (audioElement.paused) {
+        await audioElement.play();
+      } else {
+        audioElement.pause();
+      }
+    } catch (error) {
+      const { handleError } = this.props;
+      handleError(error.message, "Navidrome playback error");
+    }
+  };
+
+  seekToPositionMs = (msTimecode: number): void => {
+    const audioElement = this.audioRef.current;
+    if (audioElement) {
+      audioElement.currentTime = msTimecode / 1000;
+    }
+  };
+
+  canSearchAndPlayTracks = (): boolean => {
+    const { navidromeAuth: navidromeUser = undefined } = this.context;
+    return NavidromePlayer.hasPermissions(navidromeUser);
+  };
+
+  datasourceRecordsListens = (): boolean => {
+    return false; // record listens
+  };
+
+  setAudioSrc = (audioElement: HTMLAudioElement, src: string): void => {
+    // eslint-disable-next-line no-param-reassign
+    audioElement.src = src;
+  };
+
+  updateVolume = (): void => {
+    const { volume = 100 } = this.props;
+    const audioElement = this.audioRef.current;
+    if (audioElement) {
+      const safeVolume = Number.isFinite(volume) ? volume : 100;
+      audioElement.volume = safeVolume / 100;
+    }
+  };
+
+  pauseAudio = (): void => {
+    const audioElement = this.audioRef.current;
+    if (audioElement && !audioElement.paused) {
```

Comment:
Reading the [documentation](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/pause):
>  if the media is already in a paused state this method will have no effect.

```suggestion
    if (audioElement) {
```